src/Google.Bigquery.V2/InsertRow.cs

```diff
@@ -170,16 +189,37 @@ private static object ConvertRowValue(object value)
                 DateTimeOffset dto = (DateTimeOffset)value;
                 return dto.ToUniversalTime().ToString("yyyy-MM-dd HH:mm:ss.FFFFFF'Z'", CultureInfo.InvariantCulture);
             }
-            // Anything else should be fine as it is. We've already validated that it's a known type.
-            return value;
+            if (value is InsertRow)
+            {
+                return ((InsertRow)value).GetJsonValues();
+            }
+            if (ValidSingleTypes.Contains(value.GetType()))
+            {
+                // Anything single value should be fine as it is. We've already validated that it's a known type.
+                return value;
+            }
+            // Note: not IEnumerable<object> as we need to box value types.
+            IEnumerable values = (IEnumerable)value;
+            return values.Cast<object>().Select(ConvertRowValue).ToArray();
         }
 
         private void ValidateValue(object value, string paramName)
         {
-            if (value != null && !ValidTypes.Contains(value.GetType()))
+            if (value == null)
+            {
+                return;
+            }
+            Type type = value.GetType();
+            if (ValidSingleTypes.Contains(type))
+            {
+                return;
+            }
+            TypeInfo typeInfo = type.GetTypeInfo();
+            if (ValidRepeatedTypes.Any(ti => ti.IsAssignableFrom(typeInfo)))
```

Comment:
Okay, have now tested - null values are ignored. I think it's reasonable not to support `List<int?>` etc at that point. I've updated the test to include a null tag and show that it's not included in the results, and I've updated the docs.